src/modules/.test.js

```diff
@@ -0,0 +1,156 @@
+//add function
+export const addTodoFormHandler = (e) => {
+  e.preventDefault();
+  const todosArray = retrieveData();
+  const newTodo = new Todo(
+    addTodoForm.elements['add-task'].value,
+    false,
+    todosArray.length + 1,
+  );
+  Todo.addTodo(newTodo);
+  const todoElement = desplayElement(newTodo);
+  appendToDOM(todoElement);
+  addTodoForm.reset();
+};
+
+
+describe('addTodoFormHandler', () => {
+  let todosArray;
+  let addTodoForm;
+
+  beforeEach(() => {
+    todosArray = [      { task: 'Task 1', completed: false, id: 1 },     
+     { task: 'Task 2', completed: false, id: 2 },    ];
+    addTodoForm = {
+      elements: {
+        'add-task': {
+          value: 'Task 3'
+        }
+      },
+      reset: jest.fn()
+    };
+    Todo.addTodo = jest.fn();
+    desplayElement = jest.fn();
+    appendToDOM = jest.fn();
+    retrieveData = jest.fn(() => todosArray);
+  });
+
+  it('should prevent default behavior of the form', () => {
+    const e = { preventDefault: jest.fn() };
+    addTodoFormHandler(e);
+    expect(e.preventDefault).toHaveBeenCalled();
+  });
+
+  it('should retrieve todos from data storage', () => {
+    addTodoFormHandler({ preventDefault: jest.fn() });
+    expect(retrieveData).toHaveBeenCalled();
+  });
+
+  it('should add a new todo to the todosArray', () => {
+    addTodoFormHandler({ preventDefault: jest.fn() });
+    expect(Todo.addTodo).toHaveBeenCalledWith({
+      task: 'Task 3',
+      completed: false,
+      id: 3
+    });
+  });
+
+  it('should display the new todo element', () => {
+    addTodoFormHandler({ preventDefault: jest.fn() });
+    expect(desplayElement).toHaveBeenCalledWith({
+      task: 'Task 3',
+      completed: false,
+      id: 3
+    });
+  });
+
+  it('should append the new todo element to the DOM', () => {
+    addTodoFormHandler({ preventDefault: jest.fn() });
+    expect(appendToDOM).toHaveBeenCalledWith(desplayElement());
+  });
+
+  it('should reset the form', () => {
+    addTodoFormHandler({ preventDefault: jest.fn() });
+    expect(addTodoForm.reset).toHaveBeenCalled();
+  });
+});
+
+
+//delete function
+const deletEventHandler = (e) => {
```

Comment:
There is one test that not passing because you are creating a function here and calling it from another file as well on line 131. Kindly fix your test. 

<img width="935" alt="Screenshot 2023-02-09 at 7 36 40 PM" src="https://user-images.githubusercontent.com/25887253/217879449-b59127ad-5d8f-4a17-ba1b-679af8cda029.png">